includes/polling/ports.inc.php

```diff
@@ -210,8 +210,10 @@
             $port_stats = snmpwalk_cache_oid($device, 'ifXEntry', $port_stats, 'IF-MIB');
         }
         $hc_test = array_slice($port_stats, 0, 1);
-        if ((!isset($hc_test[0]['ifHCInOctets']) && !is_numeric($hc_test[0]['ifHCInOctets'])) ||
-            ((!isset($hc_test[0]['ifHighSpeed']) && !is_numeric($hc_test[0]['ifHighSpeed'])))) {
+        if (!isset($hc_test[0]['ifHCInOctets'])    || !is_numeric($hc_test[0]['ifHCInOctets'])    ||
+            !isset($hc_test[0]['ifHighSpeed'])     || !is_numeric($hc_test[0]['ifHighSpeed'])     ||
+            !isset($hc_test[0]['ifHCInUcastPkts']) || !is_numeric($hc_test[0]['ifHCInUcastPkts']) ||
+            $hc_test[0]['ifHCInUcastPkts'] <= 0) {
```

Comment:
ifHCInUcastPkts could be legitimately zero so not sure `<=` is the best check.

Reply:
Hmm, yes, if it's zero it will be ignored, but ifEntry could still get loaded in full on devices that don't need it. I could see about making a per-model flag, but I couldn't know the exact list of devices that need it.

Of course, I suppose there could be a user-configurable per-device for option for skipping the optimized ifEntry loading, that would fix the problem on these.

Reply:
Is this issue not a firmware issue? I can't help but feel like we're putting more work arounds when this should be fixed (if it already isn't) by the vendor.

Reply:
@laf that's one way of looking at it. But currently libremns is explicitly made to read only the 32-bit counters, which doesn't cause just missing data, but broken data (because of the overflows). In my opinion, missing packet counters is better than broken octet counters, so the least-nuisance solution would be to just the read the HC counters as usual, and ignore the fact that the packet counters show zeroes. Getting zeroes should be rather obvious, and immediately noticeable, which overflowing counters might not be. So just remove the `bad_ifXEntry` then, and avoid throwing the baby out with the bathwater.